src/unix/cygwin/mod.rs

```diff
@@ -0,0 +1,3137 @@
+
+    pub struct termios {
+        pub c_iflag: tcflag_t,
+        pub c_oflag: tcflag_t,
+        pub c_cflag: tcflag_t,
+        pub c_lflag: tcflag_t,
+        pub c_line: c_char,
+        pub c_cc: [cc_t; NCCS],
+        pub c_ispeed: speed_t,
+        pub c_ospeed: speed_t,
+    }
+
+    pub struct sched_param {
+        pub sched_priority: c_int,
+    }
+
+    pub struct flock {
+        pub l_type: c_short,
+        pub l_whence: c_short,
+        pub l_start: off_t,
+        pub l_len: off_t,
+        pub l_pid: pid_t,
+    }
+}
+
+s_no_extra_traits! {
+    #[allow(missing_debug_implementations)]
+    #[repr(align(16))]
+    pub struct max_align_t {
+        priv_: [f64; 4],
+    }
+
+    pub struct fd_set {
+        fds_bits: [fd_mask; FD_SETSIZE / ULONG_SIZE],
+    }
+
+    pub struct _uc_fpxreg {
+        pub significand: [u16; 4],
+        pub exponent: u16,
+        pub padding: [u16; 3],
+    }
+
+    pub struct _uc_xmmreg {
+        pub element: [u32; 4],
+    }
+
+    pub struct _fpstate {
+        pub cwd: u16,
+        pub swd: u16,
+        pub ftw: u16,
+        pub fop: u16,
+        pub rip: u64,
+        pub rdp: u64,
+        pub mxcsr: u32,
+        pub mxcr_mask: u32,
+        pub st: [_uc_fpxreg; 8],
+        pub xmm: [_uc_xmmreg; 16],
+        pub padding: [u32; 24],
+    }
+
+    #[repr(align(16))]
+    pub struct mcontext_t {
+        pub p1home: u64,
+        pub p2home: u64,
+        pub p3home: u64,
+        pub p4home: u64,
+        pub p5home: u64,
+        pub p6home: u64,
+        pub ctxflags: u32,
+        pub mxcsr: u32,
+        pub cs: u16,
+        pub ds: u16,
+        pub es: u16,
+        pub fs: u16,
+        pub gs: u16,
+        pub ss: u16,
+        pub eflags: u32,
+        pub dr0: u64,
+        pub dr1: u64,
+        pub dr2: u64,
+        pub dr3: u64,
+        pub dr6: u64,
+        pub dr7: u64,
+        pub rax: u64,
+        pub rcx: u64,
+        pub rdx: u64,
+        pub rbx: u64,
+        pub rsp: u64,
+        pub rbp: u64,
+        pub rsi: u64,
+        pub rdi: u64,
+        pub r8: u64,
+        pub r9: u64,
+        pub r10: u64,
+        pub r11: u64,
+        pub r12: u64,
+        pub r13: u64,
+        pub r14: u64,
+        pub r15: u64,
+        pub rip: u64,
+        pub fpregs: _fpstate,
+        pub vregs: [u64; 52],
+        pub vcx: u64,
+        pub dbc: u64,
+        pub btr: u64,
+        pub bfr: u64,
+        pub etr: u64,
+        pub efr: u64,
+        pub oldmask: u64,
+        pub cr2: u64,
+    }
+
+    pub struct sigevent {
+        pub sigev_value: sigval,
+        pub sigev_signo: c_int,
+        pub sigev_notify: c_int,
+        __unused1: *mut c_void, //actually a function pointer
+        pub sigev_notify_attributes: *mut pthread_attr_t,
+    }
+
+    pub struct siginfo_t {
+        pub si_signo: c_int,
+        pub si_code: c_int,
+        pub si_pid: pid_t,
+        pub si_uid: uid_t,
+        pub si_errno: c_int,
+        __pad: [u32; 32],
+    }
```

Comment:
Do you have a link to the header this is defined in? It doesn't seem to match https://github.com/cygwin/cygwin/blob/4bcc6adec765ee8354bfc9e6c060c9d1c0c7bf46/newlib/libc/include/sys/signal.h#L69-L73.

Reply:
https://github.com/cygwin/cygwin/blob/main/winsup/cygwin/include/cygwin/signal.h#L158-L203